### PR TITLE
feat: Admin API for deleting access token of a client

### PR DESCRIPTION
**Pull request by ajanthan**
## Related issue
#1728

## Proposed changes

Introduced a new admin API for deleting tokens of a client. The API is in the following format

`DELETE <hydra-admin>/oauth2/tokens?client_id=...`

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security. vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [x] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added or changed [the documentation](docs/docs).

## Further comments

I haven't generated the documentation since my local copy was changing the format of the existing contents. If that is ok I can run the doc gen task and commit the generated docs.

### CODE REVIEW

**Review by zepatrik — CHANGES_REQUESTED** [review]
Thank you, this looks good :+1: 
I think the only thing that's missing is tests for the storage.

**Comment by ajanthan** [comment]
> Thank you, this looks good 👍
> I think the only thing that's missing is tests for the storage.

Added a new test for the storage

**Review by zepatrik — APPROVED** [review]
Thank you

File: oauth2/handler.go [hunk 1]
```diff
@@ -726,6 +727,37 @@ func (h *Handler) forwardError(w http.ResponseWriter, r *http.Request, err error
 	http.Redirect(w, r, urlx.CopyWithQuery(h.c.ErrorURL(), query).String(), http.StatusFound)
 }
 
+// swagger:route DELETE /oauth2/token admin deleteOAuth2Token
```
Comment (aeneasr) [inline oauth2/handler.go:730]: I would prefer to not re-use the OAuth2 endpoint here but instead make it an explicit REST resource by using the plural:

```suggestion
// swagger:route DELETE /oauth2/tokens admin deleteOAuth2Token
```

What do you think?

Comment (ajanthan) [reply oauth2/handler.go]: +1

**Review by aeneasr — APPROVED** [review]
Awesome 🎉 

Thank you for your contribution!

**Pull request merged**